baybe/exceptions.py

```diff
@@ -45,4 +45,8 @@ class NumericalUnderflowError(Exception):
 
 
 class OptionalImportError(Exception):
-    """Attempted to import an optional but uninstalled dependency."""
+    """An attempt was made to import an optional but uninstalled dependency."""
+
+
+class DeprecationError(Exception):
+    """Can be used to signal deprecations to the user, interrupting code exception."""
```

Comment:
This does not fit with the other exceptions. It should rather be "Signals a the use of a deprecated aspect to the user, interrupting code execution".

Reply:
Thanks, was looking exactly for something like that. Will change 👍🏼